### DISCUSSION TITLE
Executor override afterExecute no TraceId

### DISCUSSION THREAD

**Discussion by qq31758581** [apache/skywalking#11393] [Q&A] [answered]
private Executor executor = new ThreadPoolExecutor(0, Integer.MAX_VALUE,
            60L, TimeUnit.SECONDS,
            new SynchronousQueue<Runnable>(),
            new NamedThreadFactory(true, "trace")) {

        @Override
        protected void afterExecute(Runnable r, Throwable t) {
            // this line no traceId
            log.error("no traceId", t);
        }

    };

![image](https://github.com/apache/skywalking/assets/19142582/d4247d65-57d0-40b0-95c9-8ec6185e672f)

![image](https://github.com/apache/skywalking/assets/19142582/5cca0abf-5f7a-4143-a609-51d27c21dfdf)

**Comment by weixiang1862** [comment]
skywalking log not always has a traceId, it only occurs when you hava a tracing context in current thread. You can check trace at that time.

**Reply by weixiang1862** [reply]
Even if your have a span in `runnable run` method, it stoped after run method, `afterExecute` can not reach it context, you can ref [CallableOrRunnableInvokeInterceptor](https://github.com/apache/skywalking-java/blob/main/apm-sniffer/apm-toolkit-activation/apm-toolkit-trace-activation/src/main/java/org/apache/skywalking/apm/toolkit/activation/trace/CallableOrRunnableInvokeInterceptor.java).

**Reply by qq31758581** [reply]
How should I use it ？ [CallableOrRunnableInvokeInterceptor](https://github.com/apache/skywalking-java/blob/main/apm-sniffer/apm-toolkit-activation/apm-toolkit-trace-activation/src/main/java/org/apache/skywalking/apm/toolkit/activation/trace/CallableOrRunnableInvokeInterceptor.java)

**Reply by weixiang1862** [reply, marked as answer]
No, you can not use it. It is the code in agent, I just want explain you why the log has no trace id.

As far as I konw, the toolkit was designed to trace runnable run method, but not include afterexecute.

You can try create another span by using tracer toolkit and stop it in afterexecute. It may works for your.
[toolkit-tracer](https://skywalking.apache.org/docs/skywalking-java/next/en/setup/service-agent/java-agent/application-toolkit-tracer/)

**Reply by qq31758581** [reply]
Thanks for your time.
